lib/workers/repository/updates/generate.ts

```diff
@@ -148,9 +148,16 @@ export function generateBranchConfig(
   const useGroupSettings = hasGroupName && groupEligible;
   logger.trace(`useGroupSettings: ${useGroupSettings}`);
   let releaseTimestamp: string;
+
+  if (depTypes.size) {
+    config.depTypes = Array.from(depTypes).sort();
+  }
+
   for (const branchUpgrade of branchUpgrades) {
     let upgrade: BranchUpgradeConfig = { ...branchUpgrade };
 
+    upgrade.depTypes = config.depTypes;
```

Comment:
This is logically wrong. depTypes belong to the branch, not to the upgrade. We need to work out why it's not working as-is

Reply:
Yes, I agree. This does not belong to the upgrade but to the config as the branch.

For what I can see all the `template.compile` are using the upgrade instead of values from the branch or upgrades, so it does not have access to it.

An approach to it could be, that we pass to the compile something like this...

```bash
      const extra = template.compile(upg.commitMessageExtra, {
        ...config,
        ...upg
      });
```

Or cherry-pick what we pass to the compile more selectively to be available for templating.

Reply:
Here @viceice :) 

I am happy to do the change, just checking if there are any other ideas or prefer going the implemented way :)

Reply:
Hey @viceice and @rarkins, checking what you would like me to do over here :)

Reply:
I'm not happy with this approach, but I think it's due to some technical debt we already have, and that this PR works around it.